src/state/DummyStateServer.cpp

```diff
@@ -49,38 +49,34 @@ void DummyStateServer::start()
     // To do this we run the server in a separate thread, forcing it to
     // have a localhost IP, then the main thread is the "client" with a
     // junk IP.
-
-    serverThread = std::thread([this] {
-        const std::shared_ptr<spdlog::logger>& logger =
-          faabric::util::getLogger();
-
-        // Override the host endpoint for the server thread. Must be localhost
-        faabric::util::getSystemConfig().endpointHost = LOCALHOST;
-
-        // Master the dummy data in this thread
-        if (!dummyData.empty()) {
-            const std::shared_ptr<state::StateKeyValue>& kv =
-              remoteState.getKV(dummyUser, dummyKey, dummyData.size());
-            std::shared_ptr<InMemoryStateKeyValue> inMemKv =
-              std::static_pointer_cast<InMemoryStateKeyValue>(kv);
-
-            // Check this kv "thinks" it's master
-            if (!inMemKv->isMaster()) {
-                logger->error("Dummy state server not master for data");
-                throw std::runtime_error("Remote state server failed");
-            }
-
-            // Set the data
-            kv->set(dummyData.data());
-            logger->debug(
-              "Finished setting master for test {}/{}", kv->user, kv->key);
+    const std::shared_ptr<spdlog::logger>& logger = faabric::util::getLogger();
```

Comment:
We need not to start the server on another thread, as it runs in the background by default.